pkg/store/consul/auditlogstore/consul_store.go

```diff
@@ -0,0 +1,126 @@
+package auditlogstore
+
+import (
+	"encoding/json"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/square/p2/pkg/audit"
+	"github.com/square/p2/pkg/store/consul/transaction"
+	"github.com/square/p2/pkg/util"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/pborman/uuid"
+)
+
+const auditLogTree string = "audit_logs"
+
+// ConsulStore is responsible for storage and retrieval of AuditLog JSON
+// objects. It operates a little differently than the other consul store types
+// in this project because the creation of an audit record is always expected
+// to be a part of a transaction. As a result, it doesn't actually need access
+// to a consul client to create records, it will simply save them to a
+// transaction object.
+type ConsulStore struct {
+	consulKV ConsulKV
+}
+
+type ConsulKV interface {
+	List(prefix string, opts *api.QueryOptions) (api.KVPairs, *api.QueryMeta, error)
+}
+
+func NewConsulStore(consulKV ConsulKV) ConsulStore {
+	return ConsulStore{
+		consulKV: consulKV,
+	}
+}
+
+func (ConsulStore) Create(
+	txn *transaction.Tx,
```

Comment:
Call it premature generality, but I wonder about replacing the consul transaction type with a context.Context and storing the TXN using the .WithValue function. It's more abstract and might make testing or otherwise using this code easier. The drawback is pulling the TXN out of the object is slightly clumsy and we lose compiler legibility wrt the TXN being provided to all functions of this type.